luigi/worker.py

```diff
@@ -971,10 +971,18 @@ def _handle_next_task(self):
                     self.add(t)
                 new_deps = [t.task_id for t in new_req]
 
+            try:
+                expl_json = json.dumps(expl)
+            except UnicodeDecodeError:
+                logger.info('Unicode decode error occur during %s task metadata serialisation, '
+                            'falling back to latin1 encoding.', task_id)
+
+                expl_json = json.dumps(expl, encoding='latin1')
```

Comment:
Perhaps keep the the idea of not crashing by simply returning a string like `"Internal Parsing Error"` as the expl string instead of doing something like assuming latin1 which I suppose can also crash.